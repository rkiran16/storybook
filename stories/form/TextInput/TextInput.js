import { html } from "lit-html";

export const TextInput = ({ label, isInvalid, disabled, size, showHelper, icon, placeholderText }) => {

  const icons = {
    open: 'expand_less',
    close: 'expand_more',
    openClose: 'unfold_more',
    tooltip: 'info_outline',
    search: 'search',
    loading: 'loading',
  }

  const helperHTML = html`<div id="emailHelp" class="form-text">Helper Text</div>`;
  const iconHTML = icon === 'loading' ? 
  html`<div class="spinner-border spinner-border-sm text-primary" role="status">
  <span class="visually-hidden">Loading...</span></div>`
  : 
  html`<span class="material-icons inline-icon--right">${icons[icon]}</span>`;
  const invalidHTML = html`<div class="invalid-feedback"><span class="material-icons md-16 me-2">info_outline</span>Error message goes here.</div>`

  const iconClass = icon ? 'form-group--icon' : '';
  const sizeClass = size !== "default" ? `form-control-${size}` : '';
  const invalidClass = isInvalid ? 'is-invalid' : '';
  const inputClasses = `form-control ${sizeClass} ${invalidClass}`

  return html`
      <div class="form-group ${iconClass}">
        <input type="text"
        class=${inputClasses} 
        ?disabled=${disabled} id="floatingInput" placeholder="${placeholderText}">
        <label for="floatingInput">${label}</label>
        ${iconClass ? iconHTML : ''}
        ${isInvalid ? invalidHTML : ''}
        ${showHelper ? helperHTML : ''}
      </div>
  `
}