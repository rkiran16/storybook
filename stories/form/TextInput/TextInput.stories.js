import { TextInput } from "./TextInput";

export default {
  title: 'Forms/TextInput',
  parameters: {
    zeplinLink: "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/62a9fb01c9b9111d8a460042",
  },
  argTypes: {
    label: {
      type: {name: 'string', required: false},
      description: "Label for text input",
      defaultValue: 'Label',
      table: {
        type: { summary: 'string' },
        defaultValue: { summary: "Label" },
      },
    },
    icon: {
      control: { type: 'radio' },
      description: "Select which icon appears, if any",
      table: {
        type: { summary: ['open', 'close', 'openClose', 'loading', 'search', 'tooltip', 'none'] },
        defaultValue: { summary: '' },
      },
      options: ['open', 'close', 'openClose', 'loading', 'search', 'tooltip', 'none'],
    },
      isInvalid: {
        type: { name: 'boolean', required: false},
        description: "Set Error State",
        defaultValue: false,
        table: {
          type: { summary: 'boolean' },
          defaultValue: { summary: false },
        },
        control: {
          type: 'boolean'
        }
      },
      disabled: {
        type: { name: 'boolean', required: false},
        description: "Use when the user cannot proceed or until an input is collected",
        defaultValue: false,
        table: {
          type: { summary: 'boolean' },
          defaultValue: { summary: false },
        },
        control: {
          type: 'boolean'
        }
      },
      size: {
        control: { type: 'radio' },
        description: "Define the size of a button",
        table: {
          type: { summary: ["sm" , "lg"] },
          defaultValue: { summary: 'default' },
        },
        options: ['sm', 'lg', 'default'],
      },
      showHelper: {
        type: { name: 'boolean', required: false},
        description: "Show helper Text",
        defaultValue: false,
        table: {
          type: { summary: 'boolean' },
          defaultValue: { summary: false },
        },
        control: {
          type: 'boolean'
        }
      },
      placeholderText: {
        type: { name: 'string', required: false}, 
        description: "Label for placeholder",
        defaultValue: 'Placeholder Text',
        table: {
          type: { summary: 'string' },
          defaultValue: { summary: "Placeholder Text" },
        }
      }
  }
};

const TextInputTemplate = args => TextInput(args);

export const Standard = TextInputTemplate.bind({});
Standard.parameters = {
  layout: 'centered',
};

export const ErrorState = TextInputTemplate.bind({});
ErrorState.args = {
  isInvalid: true
}

export const Disabled = TextInputTemplate.bind({});
Disabled.args = {
   disabled: true
}

export const HelperState = TextInputTemplate.bind({});
HelperState.args = {
  showHelper: true
}