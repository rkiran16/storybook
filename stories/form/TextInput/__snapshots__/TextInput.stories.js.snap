// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Forms/TextInput Disabled 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
      <div class=\\"form-group ",
    "\\">
        <input type=\\"text\\"
        class=",
    " 
        ?disabled=",
    " id=\\"floatingInput\\" placeholder=\\"",
    "\\">
        <label for=\\"floatingInput\\">",
    "</label>
        ",
    "
        ",
    "
        ",
    "
      </div>
  ",
  ],
  "values": Array [
    "",
    "form-control form-control-undefined ",
    true,
    "Placeholder Text",
    "Label",
    "",
    "",
    "",
  ],
}
`;

exports[`Storyshots Forms/TextInput Error State 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
      <div class=\\"form-group ",
    "\\">
        <input type=\\"text\\"
        class=",
    " 
        ?disabled=",
    " id=\\"floatingInput\\" placeholder=\\"",
    "\\">
        <label for=\\"floatingInput\\">",
    "</label>
        ",
    "
        ",
    "
        ",
    "
      </div>
  ",
  ],
  "values": Array [
    "",
    "form-control form-control-undefined is-invalid",
    false,
    "Placeholder Text",
    "Label",
    "",
    Object {
      "_$litType$": 1,
      "strings": Array [
        "<div class=\\"invalid-feedback\\"><span class=\\"material-icons md-16 me-2\\">info_outline</span>Error message goes here.</div>",
      ],
      "values": Array [],
    },
    "",
  ],
}
`;

exports[`Storyshots Forms/TextInput Helper State 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
      <div class=\\"form-group ",
    "\\">
        <input type=\\"text\\"
        class=",
    " 
        ?disabled=",
    " id=\\"floatingInput\\" placeholder=\\"",
    "\\">
        <label for=\\"floatingInput\\">",
    "</label>
        ",
    "
        ",
    "
        ",
    "
      </div>
  ",
  ],
  "values": Array [
    "",
    "form-control form-control-undefined ",
    false,
    "Placeholder Text",
    "Label",
    "",
    "",
    Object {
      "_$litType$": 1,
      "strings": Array [
        "<div id=\\"emailHelp\\" class=\\"form-text\\">Helper Text</div>",
      ],
      "values": Array [],
    },
  ],
}
`;

exports[`Storyshots Forms/TextInput Standard 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
      <div class=\\"form-group ",
    "\\">
        <input type=\\"text\\"
        class=",
    " 
        ?disabled=",
    " id=\\"floatingInput\\" placeholder=\\"",
    "\\">
        <label for=\\"floatingInput\\">",
    "</label>
        ",
    "
        ",
    "
        ",
    "
      </div>
  ",
  ],
  "values": Array [
    "",
    "form-control form-control-undefined ",
    false,
    "Placeholder Text",
    "Label",
    "",
    "",
    "",
  ],
}
`;
